plugins/@grouparoo/google-sheets/src/lib/shared/Spreadsheet.ts

```diff
@@ -269,6 +270,32 @@ export default class Spreadsheet {
     }
   }
 
+  async addRowsAtTheEnd(payload: any[], primaryKey) {
+    await this.load();
+    let objectWithAllKeys = {};
+    for (const entry of payload) {
+      objectWithAllKeys = Object.assign(objectWithAllKeys, entry);
+    }
+    await this.ensureHeaders(objectWithAllKeys);
+    try {
+      const addedRows = await this.sheet.addRows(payload);
+      if (addedRows) {
+        return addedRows.map((row) => ({
+          primaryKey: row[primaryKey],
+          destinationId: row._rowNumber,
+        }));
+      }
+      return null;
+    } catch (error) {
+      if (error?.response?.status === 429) {
+        await this.sleep(WAIT_TIME);
+        return this.addRowAtTheEnd(payload);
```

Comment:
this looks wrong. missing the `s` (`addRowsAtTheEnd`) and also pass `primaryKey`